metricbeat/module/prometheus/collector/collector.go

```diff
@@ -140,3 +154,58 @@ func (m *MetricSet) addUpEvent(eventList map[string]common.MapStr, up int) {
 	}
 
 }
+
+func (m *MetricSet) skipFamily(family *dto.MetricFamily) bool {
+	// example:
+	//	include_metrics:
+	//		- node_*
+	//	exclude_metrics:
+	//		- node_disk_*
+	//
+	// This would mean that we want to keep only the metrics that start with node_ prefix but
+	// are not related to disk so we exclude node_disk_* metrics from them.
+
+	if family == nil {
+		return true
+	}
+
+	// if include_metrics are defined, check if this metric should be included
+	if len(m.includeMetrics) > 0  {
+		if !matchMetricFamily(*family.Name, m.includeMetrics) {
+			return true
+		}
+	}
+	// now exclude the metric if it matches any of the given patterns
+	if len(m.ignoreMetrics) > 0 {
+		if matchMetricFamily(*family.Name, m.ignoreMetrics) {
+			return true
+		}
+	}
+	return false
+}
+
+func compilePatternList(patterns *[]string) []*regexp.Regexp{
+	var compiledPatterns []*regexp.Regexp
+	compiledPatterns = []*regexp.Regexp{}
+	if patterns != nil {
+		for _, pattern := range *patterns {
+			r, err := regexp.Compile(pattern)
+			if err != nil {
+				continue
```

Comment:
These errors should make `New()` to fail, if not there will be incorrect patterns silently ignored.

Reply:
hmm, ignoring them was an opinionated decision. I considered it not so important but now after thinking it again I agree we should let the user know. Thanks!

Reply:
Thanks!